rpcs3/Emu/RSX/Common/TextureUtils.cpp

```diff
@@ -525,40 +525,38 @@ u8 get_format_block_size_in_bytes(rsx::surface_color_format format)
 	}
 }
 
-size_t get_placed_texture_storage_size(u16 width, u16 height, u32 depth, u8 format, u16 mipmap, bool cubemap, size_t row_pitch_alignment, size_t mipmap_alignment)
+size_t get_placed_texture_storage_size(u16 width, u16 height, u32 depth, u8 format, u16 mipmap, rsx::texture_dimension_extended dimension, size_t row_pitch_alignment, size_t mipmap_alignment)
 {
-	size_t w = width;
-	size_t h = std::max<u16>(height, 1);
-	size_t d = std::max<u16>(depth, 1);
-
+	if (dimension != rsx::texture_dimension_extended::texture_dimension_3d) depth = 1;
```

Comment:
move this to tex.depth(), otherwise other logic may fail. I know we don't properly check for this anyway, but still, if wierd depth sizes are allowed, it should be clamped at the source.